### PR TITLE
chore: add new publish pipeline

### PR DESCRIPTION
**Pull request by rzhao271**
Also fixes up the existing pipeline to become a CI-only pipeline.
One issue with the new pipeline is that it uses the package.json version. I'm not sure how to let it auto-generate a new beta version.

### CODE REVIEW

**Comment by Tyriar** [comment]
> One issue with the new pipeline is that it uses the package.json version. I'm not sure how to let it auto-generate a new beta version.

It would be a shame to lose that, would it instead publish whenever the package.json changes in main?

**Comment by rzhao271** [comment]
I have changed the publish script to only increment the package.json version, and the YAML should now run that script while building the package before publishing it. There is no verification build because the pipeline needs to be on the main branch for me to create a new pipeline in ADO for it.

**Comment by lszomoru** [comment]
You should look at `vsce` as it implements the desired pattern to automatically update the package version based on the `nextVersion` pipeline input parameter - https://github.com/microsoft/vscode-vsce/blob/f1fc6d255d634849db9a7f28e59a031c49cd4918/build/azure-pipelines.yml#L18

**Comment by rzhao271** [comment]
@lszomoru I believe the version format is different for node-pty: https://www.npmjs.com/package/node-pty?activeTab=versions

Here's what I see for `@vscode/vsce`: https://www.npmjs.com/package/@vscode/vsce?activeTab=versions

**Comment by lszomoru** [comment]
If node-pty is using custom version number format then it is up to the package owner to increment it. If you specify the `nextVersion` template parameter (list vsce does) we will run `npm version ${{ parameters.nextVersion }} --no-git-tag-version --allow-same-version` to update the `package.json`.

File: scripts/increment-version.js [hunk 1]
```diff
@@ -7,30 +7,19 @@ const fs = require('fs');
...
 
 // Set the version in package.json
 const packageJsonFile = path.resolve(__dirname, '..', 'package.json');
 packageJson.version = nextVersion;
 fs.writeFileSync(packageJsonFile, JSON.stringify(packageJson, null, 2));
 
-// Publish
-const args = ['publish'];
-if (!isStableRelease) {
-  args.push('--tag', 'beta');
```
Comment (rzhao271) [inline scripts/increment-version.js:29]: I believe the pipeline will have to support this tag functionality as well. Leaving the PR unmerged.

Comment (lszomoru) [reply scripts/increment-version.js:29]: Yes, we automatically create a GH release and tag.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -24,97 +24,49 @@ extends:
...
             displayName: 'Test'
-          - script: |
-              npm run lint
+          - script: npm run lint
             displayName: 'Lint'
 
         - job: macOS
           pool:
             name: Azure Pipelines
             vmImage: 'macOS-latest'
```
Comment (lszomoru) [inline azure-pipelines.yml:42]: Please don't use macOS-latest, instead use an explicit version.

File: scripts/increment-version.js (hunk 1 above)
Comment (rzhao271) [reply scripts/increment-version.js:29]: I have added tag parameters to the YAML that should tell ESRP to publish with the correct tag.

**Comment by rzhao271** [comment]
There is only a verification build for the CI pipeline at this time: https://dev.azure.com/vscode/node-pty/_build/results?buildId=138802&view=results

I will create and test the publishing pipeline _after_ this PR is merged. Afaik, I am unable to create the pipeline using the Start Right process from a non-main branch.

**Review by lszomoru — APPROVED** [review]

**Review by Tyriar — APPROVED** [review]

**Pull request merged**